app/scripts/views/settings/recovery_codes.js

```diff
@@ -97,6 +97,19 @@ const View = FormView.extend({
   },
 
   setInitialContext(context) {
+    // if user has no totp setup then the user should not be able to
+    // directly navigate to /settings/two_step_authentication/recovery_codes
+    const that = this;
+    const account = this.getSignedInAccount();
+
+    account.checkTotpTokenExists()
```

Comment:
I would pull this logic into a `beforeRender` function. This function gets called automatically to setup required things before the view is show. Example, https://github.com/mozilla/fxa-content-server/blob/58b8eaac1819f490be9c5b90b7d3d4b72742268e/app/scripts/views/reset_password.js#L45.